### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by jackcarlisle**
Set up of test runner, coverage, linter, precommit hooks, travis yml....

Basically bare minimum but will continue with hapi specific stuff in https://github.com/o2tank/hapi-starter-kit

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+coverage
```
Comment (jrans) [inline .eslintignore]: nope, and not sure it is in its remit? I mean we could be ignoring lots of files produced by other npm modules but where do we draw the line?

Comment (nelsonic) [inline .eslintignore]: `/coverage` is one we have in _all_ our projects. I can't think of a situation where we _wouldn't_ want to ignore it. not really bothered about any others e.g: `/build` or `/dist` ...

Comment (jrans) [inline .eslintignore]: ok will raise an issue in goodparts!

File: .istanbul.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+verbose: false
```
Comment (jrans) [inline .istanbul.yml]: will change

**Comment by iteles** [comment]
Assigned back to @jrans for changes noted above.

File: .eslintignore (hunk 1 above)
Comment (jrans) [inline .eslintignore]: https://github.com/dwyl/goodparts/issues/255

**Comment by nelsonic** [comment]
@jackcarlisle + @jrans looks good. merging. 👍 
(_hope you both have a good weekend!_)

**Pull request merged**